### PR TITLE
feat: AR-1294 update self handle use case

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

AR-1294

### Testing

unit tests

----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by typfel — APPROVED** [review]
LGMT I had some comments which are not necessarily related to this PR

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt [hunk 1]
```diff
@@ -46,24 +50,30 @@ class UserDataSource(
...
             // Fetching self user team here because SyncSelfUserUseCase is (which is using this function)
             // doesn't have a return value, and this team fetch doesn't make sense adding to GetSelfUserUseCase
             // as we don't want to be fetching the user team everytime we get the user.
-            user.team?.let { teamId -> teamRepository.fetchTeamById(teamId = teamId) }
-            assetRepository.downloadUsersPictureAssets(listOf(user.previewAssetId, user.completeAssetId))
-            userDAO.insertUser(user)
-            metadataDAO.insertValue(Json.encodeToString(user.id), SELF_USER_ID_KEY)
+            userEntity.team?.let { teamId -> teamRepository.fetchTeamById(teamId = teamId) }
+            assetRepository.downloadUsersPictureAssets(listOf(userEntity.previewAssetId, userEntity.completeAssetId))
+            // TODO: handle storage error
```
[the remark this answers is not in the document]
Comment (MohamadJaara) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt]: Any exceptions the `persistence` module can throw, i already have a PR for it #275

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt [hunk 2]
```diff
@@ -46,24 +50,30 @@ class UserDataSource(
     private val teamRepository: TeamRepository
 ) : UserRepository {
 
+    private suspend fun getSelfUserId(): QualifiedIDEntity {
```
[the remark this answers is not in the document]
Comment (MohamadJaara) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt:51]: No, it's not renamed, but there are 3 `QualifiedID` in logic, storage, and network
and in the import 
```kotlin
import com.wire.kalium.persistence.dao.QualifiedID as QualifiedIDEntity
```

[the remark this answers is not in the document]
Comment (MohamadJaara) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt:51]: We need some kind of shared module that that all module depends on if we are going to share models like IDs

**Pull request merged**